lib/Doctrine/ORM/Query/QueryException.php

```diff
@@ -75,28 +75,6 @@ public static function invalidParameterPosition($pos)
         return new self('Invalid parameter position: ' . $pos);
     }
 
-    /**
-     * @param int $expected
-     * @param int $received
-     *
-     * @return QueryException
-     */
-    public static function tooManyParameters($expected, $received)
```

Comment:
@greg0ire exceptions are your thing: how much does this overlap with your patch?

Reply:
It doesn't: I restrained myself to ORMException for now.

Reply:
Thanks!

@svencan besides improvements to be applied as per https://github.com/doctrine/doctrine2/pull/7084#discussion_r170472606, this looks good :+1: